src/prophecy/layer/dense_layer.hh

```diff
@@ -47,7 +53,16 @@ void DenseLayer<T>::compile(std::weak_ptr<Layer<T>>   prev,
     // Initialize weights and delta_weights
     const std::vector<uint> w_shape = {this->nb_neurons_,
                                        prev.lock()->get_out_shape()[0]};
-    this->weights_                  = xt::random::randn<T>(w_shape, 0, 1);
+    if (init_ == "xavier")
+    {
+        const T weights_std = sqrt(((T)2) / (w_shape.front(), w_shape.back()));
+        this->weights_              = xt::random::randn<T>(w_shape, 0,
+                                                           weights_std);
+    }
+    else
+    {
+        this->weights_              = xt::random::randn<T>(w_shape, 0, 1);
+    }
     this->delta_weights_            = xt::zeros<T>(w_shape);
 
     // Initialize biases and delta_biases
```

Comment:
According to the original implementation of Xavier/Glorot initialization (http://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf), biases should be set to 0.